src/scenes/ApplicationRouter/provider/index.tsx

```diff
@@ -50,20 +55,63 @@ const ApplicationProvider = ({ children }: Props) => {
     [allApplications]
   );
 
-  const _reload = useCallback(() => {
+  const _reload = useCallback(async () => {
+    if (selectedApplication) {
+      const result = await ApplicationService.getApplication(selectedApplication.id);
```

Comment:
Should this go into redux and our normal request flow?

Reply:
This is done this way because a navigate will occur before reload is completed without this synchronized step.

I am hoping to actually not have to fetch one off application at all if possible. `requestListApplications` will end up refetching the data. I think at some point I will need to add some indication of `isLoading` in the provider. But one for the later.

Reply:
I see, if you use redux for this, you can just select the application out of state using the ID. This would allow you to select a single application out of the "list applications" redux state.

Reply:
Good call here! I can likely keep track of selectedApplicationId, and set selectedApplication to null for loading state.

Reply:
Tracking this task as a future upgrade!